### PR TITLE
♻️Rename ShadowDomWriter* => DomWriter*

### PR DESCRIPTION
**Pull request by calebcordry**
Rename classes and move to their own file in preparation for ads use-case. After reviewing again, I don't think we will need to change the impl (not really anything shadow specific) but if necessary will do in a follow up PR.

### CODE REVIEW

File: src/dom-writer.js [hunk 1]
```diff
@@ -0,0 +1,349 @@
+/**
```
Comment (dvoytenko) [inline src/dom-writer.js]: Should we move this to `utils/`?

Comment (calebcordry) [reply src/dom-writer.js]: Sure, done.

**Comment by calebcordry** [comment]
Looking into test failures, I can repro on clean master so I think something strange is going on.

**Comment by calebcordry** [comment]
Test failures should be fixed by #26607 when merged 🤞

**Pull request merged**